bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyEventServlet.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.shelly.internal.api;
+
+import static org.openhab.binding.shelly.internal.ShellyBindingConstants.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.shelly.internal.ShellyHandlerFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link ShellyEventServlet} implements a servlet. which is called by the Shelly device to signnal events (button,
+ * relay output, sensor data). The binding automatically sets those vent urls on startup (when not disabled in the thing
+ * config).
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = HttpServlet.class, configurationPolicy = ConfigurationPolicy.OPTIONAL, immediate = true)
+public class ShellyEventServlet extends HttpServlet {
+    private static final long serialVersionUID = 549582869577534569L;
+    private final Logger logger = LoggerFactory.getLogger(ShellyEventServlet.class);
+
+    private @Nullable HttpService httpService;
+    private @Nullable ShellyHandlerFactory handlerFactory;
+
+    @SuppressWarnings("null")
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        try {
+            httpService.registerServlet(SHELLY_CALLBACK_URI, this, null, httpService.createDefaultHttpContext());
+            logger.info("Shelly: CallbackServlet started at '{}'", SHELLY_CALLBACK_URI);
+            // } catch (ServletException | NamespaceException e) {
+        } catch (RuntimeException | NamespaceException | ServletException e) {
+            logger.error("Shelly: Could not start CallbackServlet: {} ({})", e.getMessage(), e.getClass());
```

Comment:
Log the actual exception if it's an error.
```suggestion
            logger.error("Shelly: Could not start CallbackServlet: {})", e.getMessage(), e);
```

Reply:
done